juriscraper/opinions/united_states/state/ga.py

```diff
@@ -15,69 +14,59 @@ class Site(OpinionSiteLinear):
-            while parent.tag != "p":
-                parent = parent.xpath("./..")[0]
-            for item in parent.getnext().xpath("./li"):
-                text = item.text_content()
-                if text:
-                    # Extract Docket numbers
-                    dockets = re.findall(self.regex_docket, text)
-                    if not dockets:
-                        if (
-                            text
-                            == "IN RE: MOTION TO AMEND 2021-3 (Bar Rule 1.2)"
-                        ):
-                            continue
-                        else:
-                            raise InsanityException(
-                                f"Could not find docket numbers in: '{text}'"
-                            )
+        :param year: Year to scrape
+        :return: URL for the given year
+        """
+        return f"https://www.gasupreme.us/opinions/{year}-opinions/"
 
-                    # Extract name substring; I am sure this could
-                    # be done with a more slick regex, but its not
-                    # my forte...
-                    name = text
-                    for docket in dockets:
-                        name = name.replace(docket, "")
-                    name = name.lstrip(" .,")
+    def _process_html(self) -> None:
+        for link in self.html.xpath("//li/a[contains(@href, '.pdf')]"):
+            url = link.get("href")
+            title = link.text_content()
+            if not title:
+                continue
+            dockets = re.findall(r"S?\d{2}\w\d{4}", title)
+            if not dockets:
+                # Skip empty links in the HTML
+                continue
+            docket = ", ".join(dockets)
+            name = title.split(dockets[-1])[-1].strip("., ")
+            summary = link.xpath(
+                ".//parent::li/parent::ul/preceding-sibling::p[1]"
+            )[0].text_content()
+            # We've got en/em dash and hyphens.
+            date_str = (
+                summary.split("–")[0].split("—")[0].split("-")[0].strip()
+            )
+            date_summary = datetime.strptime(date_str, "%B %d, %Y")
+            if self.test_mode_enabled():
+                self.last_month = date(year=2019, month=11, day=1)
 
-                    self.cases.append(
-                        {
-                            "date": date_string,
-                            "docket": ", ".join(dockets),
-                            "name": titlecase(name.lstrip(" .,")),
-                            "url": item.xpath(".//a[1]/@href")[0],
-                        }
-                    )
+            if date_summary.date() < self.last_month:
+                # GA lists all opinions from the previous year on the same page
+                # but we don't need to scrape them every time.  So cancel out
+                # after 31 days.
+                # This should resolve the speed warning we receive on GA.
+                break
```

Comment:
This should be left to CL. It's better to give CL lots of information and let it decide when to cancel.

Reply:
Removed date restriction for the opinions to scrape.
All opinions published in the year will be scraped, regardless of their publication date.